### PR TITLE
Use trussed-auth

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -51,8 +52,10 @@ serde_cbor = "0.11"
 hex = { version = "0.4", features = ["serde"] }
 
 [features]
+default = []
 std = []
-virtual = ["std", "vpicc"]
+virtual = ["std", "vpicc", "virt"]
+virt = ["std", "trussed/virt"]
```
Comment (robin-nitrokey) [inline Cargo.toml:58]: Maybe rename `virtual` to `vpicc`?

File: src/state.rs [hunk 2]
```diff
@@ -369,129 +448,90 @@ impl Persistent {
...
-            Password::ResetCode => Self::MAX_RETRIES.saturating_sub(self.reset_code_tries),
-        }
+    pub fn remaining_tries<T: trussed::Client + AuthClient>(
+        &self,
+        client: &mut T,
+        password: Password,
+    ) -> u8 {
+        try_syscall!(client.pin_retries(password))
+            .map(|r| r.retries.unwrap_or_default())
+            .unwrap_or(Self::MAX_RETRIES)
```
Comment (robin-nitrokey) [inline src/state.rs:458]: Return 0 instead?  If `try_syscall` is `Err`, the PIN is not set for some reason, so it cannot be verified.

**Review by robin-nitrokey — COMMENTED** [review]
The changes look good to me.  But as far as I see, `init_pins` is only called if there is no valid state.  Couldn’t this lead to problems when upgrading from an older version?

File: Cargo.toml (hunk 1 above)
Comment (sosthene-nitrokey) [reply Cargo.toml:60]: Let's not add things to this PR, which is already large enough (even more true for #127 ). I'll make an issue and another PR. I'll also rename it in PIV.

File: src/state.rs (hunk 2 above)
Comment (sosthene-nitrokey) [reply src/state.rs:458]: This makes me realize that we were always returning `3` for the PUK, when `0` makes more sense, which this would fix.

**Comment by sosthene-nitrokey** [comment]
> The changes look good to me. But as far as I see, `init_pins` is only called if there is no valid state. Couldn’t this lead to problems when upgrading from an older version?

I added pin deletion to the factory reset. What do you see that would go wrong?

**Comment by robin-nitrokey** [comment]
I think the path I saw was:  Suppose you initialized the state with an old opcard version and stored the PINs in the FS.  Now you update to this version.  There are no PINs in trussed-auth, but there is a valid state.  Factory reset is only allowed if a) verified as admin or b) if the admin PIN is locked.  a) is not possible without a PIN, and b) was not possible because the retries would default to 3 if the PIN is not set.  This should be fixed now.

**Comment by robin-nitrokey** [comment]
> I added pin deletion to the factory reset.

Shouldn’t it reset the PINs to the defaults?

**Comment by sosthene-nitrokey** [comment]
On factory reset, the state and pins are deleted.

On state load, if no state is present the pins  are set to their default values.

**Review by robin-nitrokey — APPROVED** [review]

**Pull request merged**